File: Two Sum/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> hashMap = new HashMap<Integer, Integer>();
        
        for(int i = 0; i < nums.length; i++) {
            
            if(hashMap.containsKey(nums[i])) {
                return new int[]{hashMap.get(nums[i]), i};
            }
            
            hashMap.put(target - nums[i], i);
        }
        
        return null;
    }
}

File: Two Sum/TwoSum.cs
public class Solution 
{
    public int[] TwoSum(int[] nums, int target) 
    {
        var complements = new Dictionary<int, int>();
        
        for(int i = 0; i < nums.Length; i++)
        {
            var complementVal = target - nums[i];
            
            if(complements.ContainsKey(complementVal))
            {
                return new int[2] {complements[complementVal], i};
            }
            
            complements.Add(nums[i], i);
        }
        
        return new int[0];
    }
}